### PR TITLE
Slack Bot to respond very quickly to acknowledge seeing the question

### PR DESCRIPTION
**Pull request by yuhongsun96**
- Reacts to message with eyes, then removes them after the answer comes through
- Respond to /danswerbot messages as soon as processing
- Dedupe @danswerbot messages
- Reworks some parts of the flow and also gives a failure message if something goes wrong

### CODE REVIEW

File: backend/danswer/bots/slack/handlers/handle_message.py [hunk 1]
```diff
@@ -7,111 +8,107 @@
...
+            reflexion = "well_answered_postfilter" in channel_conf["answer_filters"]
+
+            if (
+                "questionmark_prefilter" in channel_conf["answer_filters"]
+                and "?" not in msg
+            ):
+                logger.info(
+                    "Skipping message since it does not contain a question mark"
+                )
+                return False
```
Comment (yuhongsun96) [inline backend/danswer/bots/slack/handlers/handle_message.py:85]: The return is now bool for if there's a failure (eg. should we tell user something has gone wrong). See if you agree with all these return True/False cases

File: backend/danswer/bots/slack/handlers/handle_message.py [hunk 2]
```diff
@@ -7,111 +8,107 @@
...
     disable_docs_only_answer: bool = DANSWER_BOT_DISABLE_DOCS_ONLY_ANSWER,
-    respond_every_channel: bool = DANSWER_BOT_RESPOND_EVERY_CHANNEL,
-) -> None:
-    engine = get_sqlalchemy_engine()
-    with Session(engine) as db_session:
-        channel_name = get_channel_name_from_id(client=client, channel_id=channel)
-        slack_bot_config = get_slack_bot_config_for_channel(
-            channel_name=channel_name, db_session=db_session
+) -> bool:
+    """Potentially respond to the user message, returns bool for failure
```
Comment (Weves) [inline backend/danswer/bots/slack/handlers/handle_message.py]: Maybe mention what this output means functionally + which True/False mean

File: backend/danswer/bots/slack/handlers/handle_message.py [hunk 3]
```diff
@@ -185,14 +182,14 @@ def _get_answer(question: QuestionRequest) -> QAResponse:
...
-        return
+        return True
 
     if not answer.answer and disable_docs_only_answer:
         logger.info(
             "Unable to find answer - not responding since the "
             "`DANSWER_BOT_DISABLE_DOCS_ONLY_ANSWER` env variable is set"
         )
-        return
+        return False
```
Comment (Weves) [inline backend/danswer/bots/slack/handlers/handle_message.py]: Make this `True` as well?

File: backend/danswer/bots/slack/listener.py [hunk 4]
```diff
@@ -105,114 +96,222 @@ def _process_slack_event(client: SocketModeClient, req: SocketModeRequest) -> No
...
                 f"Ignoring message with subtype '{message_subtype}' since is is a special message type"
             )
-            return
+            return False
 
         message_ts = event.get("ts")
         thread_ts = event.get("thread_ts")
         # Pick the root of the thread (if a thread exists)
-        message_ts_to_respond_to = cast(str, thread_ts or message_ts)
+        cast(str, thread_ts or message_ts)
```
Comment (Weves) [inline backend/danswer/bots/slack/listener.py]: I don't think this does anything?

File: backend/danswer/bots/slack/utils.py [hunk 5]
```diff
@@ -21,6 +23,20 @@
 logger = setup_logger()
 
 
+class _ChannelIdAdapter(logging.LoggerAdapter):
```
Comment (Weves) [inline backend/danswer/bots/slack/utils.py]: nit: remove `_`, if present it's meant to be private / not importable

**Review by Weves — APPROVED** [review]

**Pull request merged**